single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/SingleHostHttpClientBuilder.java

```diff
@@ -0,0 +1,185 @@
+package com.github.nhenneaux.resilienthttpclient.singlehostclient;
+
+import javax.net.ssl.SNIHostName;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import java.net.http.HttpClient;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Optional;
+
+import static com.github.nhenneaux.resilienthttpclient.singlehostclient.SingleHostHttpClientBuilder.RethrowGeneralSecurityException.handleGeneralSecurityException;
+
+
+/**
+ * Create an {@link HttpClient} to target a single host.
+ * It validates the certificate to authenticate the server in TLS communication with this single name.
+ * It can be used to target a single host using its IP address(es) instead of its hostname while keeping a high protection against Man-in-the-middle attack.
+ * <p>
+ * <code>-Djdk.internal.httpclient.disableHostnameVerification</code> is needed to use a custom TLS name matching based on the requested host instead of the one from the URL.
+ * <p>
+ * <code>-Djdk.httpclient.allowRestrictedHeaders=Host</code> is needed to customize the HTTP Host header.
+ */
+@SuppressWarnings({"WeakerAccess", "unused"}) // To use outside the module
+public class SingleHostHttpClientBuilder {
+
+    private final String hostname;
+    private final HttpClient.Builder builder;
+
+    private SingleHostHttpClientBuilder(String hostname, HttpClient.Builder builder) {
+        this.hostname = hostname;
+        this.builder = builder;
+    }
+
+    public static SingleHostHttpClientBuilder builder(String hostname) {
+        return new SingleHostHttpClientBuilder(hostname, HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(2)));
```

Comment:
Not a configurable value?

Reply:
it can be changed when the client builder is provided (https://github.com/nhenneaux/resilient-httpclient/blob/use_builder_singleipclient/single-host-httpclient/src/main/java/com/github/nhenneaux/resilienthttpclient/singlehostclient/SingleHostHttpClientBuilder.java#L44).
Otherwise it provides a more relevant value than default of HTTP client (infinite).